ipi/engine/forces.py

```diff
@@ -960,29 +960,26 @@ def transfer_forces_manual(
         assert len(self.mforces) == len(new_q), msg
         assert len(self.mforces) == len(new_v), msg
         assert len(self.mforces) == len(new_forces), msg
-        if new_x == None:
-            new_x = [[None] * self.nbeads] * len(self.mforces)
-            info("WARNING: No extras information has been passed.", verbosity.debug)
-
-        assert len(self.mforces) == len(new_x), msg
-
+        if new_extra is None:
+            new_extra = [{} for i in range(len(self.mforces))]
         for k in range(len(self.mforces)):
             mv = new_v[k]
             mf = new_forces[k]
             mq = new_q[k]
-            mextra = new_x[k]
+            mextra = new_extra[k]
             mself = self.mforces[k]
 
             assert mq.shape == mf.shape, msg
             assert mq.shape[0] == mv.shape[0], msg
             assert mself.nbeads == mv.shape[0], msg
             assert mself.nbeads == mq.shape[0], msg
-            assert mself.nbeads == len(mextra), msg
-            mxlist = mextra[:]
 
             dd(mself.beads).q.set(mq, manual=False)
             for b in range(mself.nbeads):
-                ufvx = [mv[b], mf[b], vir, mxlist[b]]
+                mx = {}
```

Comment:
The modifications around this line are not intuitive to me. Why are they needed? I am concerned it will add unnecessary overhead and could break other things down the line?

Reply:
Ok, as we discussed in slack we can leave as is, but probably need to change the arrangement of the array itself later.